forms/Validator.php

```diff
@@ -81,10 +85,12 @@ public function getErrors() {
 	}
 
 	/**
-	 * @param string $fieldName
-	 * @param array $data
+	 * Ensures the specified field is not empty.
+	 *
+	 * @param	string	$fieldName	The name of the valid you wish to ensure is not empty.
+	 * @param	array	$data		An array of field values you want to check against.
 	 */
-	public function requireField($fieldName, $data) {
+	public function requireField($fieldName, array $data) {
```

Comment:
You're right, the `is_array` check happening below makes it somewhat pointless.

 How can we ensure that developers utilizing this API are aware of their mistake ahead of time without causing crashes/issues when in production? Seems like an easy mistake to make since this method expects an array of fields despite the name being singular. I like doing this semantically at the language level (i.e. the `array` type hint) since this method will not work if this isn't an array, but I don't like crashing code should it make it into production and I also don't like the inelegant excess of calling `trigger_error()` in a minor circumstance such as this.